src/openassetio-python/package/openassetio/__init__.py

```diff
@@ -70,12 +70,20 @@
 The documentation for OpenAssetIO can be found here:
    https://openassetio.github.io/OpenAssetIO.
 """
-# TODO(DF): @pylint
-from ._openassetio import (  # pylint: disable=import-error
+
+# pylint: disable=wrong-import-position,import-error
+
+from ._openassetio import (
     constants,
-    TraitsData,
     Context,
     EntityReference,
 )
 
-# pylint: disable=wrong-import-position
+
+#
+# Deprecated: https://github.com/OpenAssetIO/OpenAssetIO/issues/1127
+#
+from ._openassetio import trait
+
+TraitsData = trait.TraitsData
```

Comment:
Had a quick investigation, and it doesn't look possible to issue deprecation warnings for an alias like this.  

There is https://peps.python.org/pep-0702 (and a [typing-extensions](https://typing-extensions.readthedocs.io/en/latest/index.html#deprecated)  shim library to bring support to current Python), but that won't work for objects, only classes and functions.

One solution would be to bind `TraitsData` twice, and shim the constructor in the old location to issue a `PyErr_WarnEx(PyExc_DeprecationWarning, "message")`.  But I think that's far too much effort and could cause weird bugs.